InfiniteLoopBookProject/tests/TestDAO.java

```diff
@@ -1,4 +1,279 @@
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 
-public class TestDAO {
+import java.io.IOException;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
 
+import com.st.lms.dao.DAOAuthorImpl;
+import com.st.lms.dao.DAOBookImpl;
+import com.st.lms.dao.DAOPublisherImpl;
+import com.st.lms.model.Author;
+import com.st.lms.model.Book;
+import com.st.lms.model.Publisher;
+
+public class TestDAO {
+	
+	@BeforeClass
+	public static void setUp() throws IOException{
+		AuthorDaoTest.setUp();
+		BookDaoTest.setUp();
+		PublisherDaoTest.setUp();
+	}
+	
+	@AfterClass
+	public static void tearDown() throws IOException{
+		AuthorDaoTest.tearDown();;
+		BookDaoTest.tearDown();;
+		PublisherDaoTest.tearDown();;
+	}
+	
+	public static void testGet() throws IOException{
+		AuthorDaoTest.testGet();
+		BookDaoTest.testGet();
+		PublisherDaoTest.testGet();
+	}
+	
+	public static void testSet() throws IOException{
+		AuthorDaoTest.testSet();
+		BookDaoTest.testSet();
+		PublisherDaoTest.testSet();
+	}
+	
+	public static void testAdd() throws IOException{
+		AuthorDaoTest.testAdd();
+		BookDaoTest.testAdd();
+		PublisherDaoTest.testAdd();
+	}
+	
+	public static void testDelete() throws IOException{
+		AuthorDaoTest.testDelete();
+		BookDaoTest.testDelete();
+		PublisherDaoTest.testDelete();
+	}
+	
+	static class AuthorDaoTest{

```

Comment:
Either `AuthorDaoTest` should be a separate (top-level) class, or its contents should just be moved to the top level.